### PR TITLE
Introduce CORS and parties CRUD test-api operations

### PR DESCRIPTION
**Pull request by ivannovazzi**
- It enables CORS for the test-api server in order to allow using it from a browser on a different host, see localhost / docker combination.
- Introduces the CRUD operations for the _parties_ on `test-api`. 
- Changes/adds the uniqueness validation on parties by checking against the idType + idValue, so that a MSIDSN and ACCOUNT_NO idType can use the same idValue.
- Fixes/adds scripts to test the test-api endpoints around creating, updating, deleting, reading parties.
- Fixes/adds script to test the simulator endpoint to read a party, participant.

### CODE REVIEW

File: src/test/test-api.js [hunk 1]
```diff
@@ -57,12 +58,59 @@ const testOps = [{
...
+    // eslint-disable-next-line no-param-reassign
+    t.context.request = { body: party };
+    // eslint-disable-next-line no-param-reassign
+    t.context.state.path = { params: { idType, idValue } };
+    await handlers.map['/repository/parties/{idType}/{idValue}'].put(t.context);
+    t.falsy(t.context.response.body);
+    t.is(t.context.response.status, 204);
+});
+
+test('should return 200 when deleting a party', async (t) => {
```
Comment (bushjames) [inline src/test/test-api.js]: should read "should retiurn 204 when deleting a party"

File: src/rules.json [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-		"params": {
-			"statusCode": 500,
-			"body": {
-                "statusCode": "4001",
-                "message": "Payer FSP insufficient liquidity"
-            }
-		}
-		}
-	}
+    "conditions": {
```
Comment (bushjames) [inline src/rules.json:5]: why the change in indentation?

File: src/index.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ const yaml = require('yamljs');
 const util = require('util');
 const router = require('@internal/router');
 const https = require('https');
-
+const cors = require('@koa/cors');
```
Comment (bushjames) [inline src/index.js:35]: why do we need CORS?

**Review by bushjames — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (ivannovazzi) [reply src/index.js:35]: To allow the the browser to interact with the backend.
Curl / postman will not complain about missing CORS, but the browser runs on a different port, so it requires CORS.

**Comment by ivannovazzi** [comment]
@bushjames I added some scripts for parties CRUD operations since the existing ones were not properly working.

File: src/rules.json (hunk 2 above)
Comment (ivannovazzi) [reply src/rules.json:5]: the original file did not have correct indentation

File: src/test/test-api.js (hunk 1 above)
Comment (ivannovazzi) [reply src/test/test-api.js]: correct, going to fix it

**Review by bushjames — APPROVED** [review]
